app/services/pricers/book_pricer.rb

```diff
@@ -0,0 +1,48 @@
+module Pricers
+  class BookPricer < Pricer
+    # @param catalog [ProductCatalog]
+    # @return [BookPricer]
+    def initialize(catalog)
+      @catalog = catalog
+    end
+
+    # @param price [Result<Float>]
+    # @param params [Hash]
+    # @param now [Time]
+    # @return [Result<Float>]
+    def compute(price, params, now)
+      return price unless params[:kind] == 'book'
```

Comment:
IIUC a pricer cannot be used standalone to price a book. It has to be call from a function in an `inject`, right?

Reply:
It's what is intended but to required.
A pricer is a rule to compute a price variation that could be combined in any way to define a calculator.
Now I'm saying this, I think names are not that good. I should rename `pricer` to `pricing_rule` or something like this.